common/ledger/blockledger/fileledger/factory.go

```diff
@@ -97,8 +109,30 @@ func New(directory string, metricsProvider metrics.Provider) (blockledger.Factor
 	if err != nil {
 		return nil, err
 	}
-	return &fileLedgerFactory{
+
+	fileRepo, err := filerepo.New(filepath.Join(directory, "filerepo"), "remove")
+	if err != nil {
+		return nil, err
+	}
+
+	factory := &fileLedgerFactory{
 		blkstorageProvider: p,
 		ledgers:            map[string]*FileLedger{},
-	}, nil
+		removeFileRepo:     fileRepo,
+	}
+
+	files, err := factory.removeFileRepo.List()
+	if len(files) != 0 {
+		for _, fileName := range files {
+			channelID := factory.removeFileRepo.FileToBaseName(fileName)
+			err = factory.Remove(channelID)
+			if err != nil {
+				logger.Errorf("error removing ledger for channel %s: %s", channelID, err.Error())
+				return nil, err
```

Comment:
I suggest
```go
logger.Errorf("Failed to remove channel %s: %s", channelID, err.Error())
```
because it is not only the ledger removal that may cause this error to be thrown.

In addition, there is no unit test coverage for when this happens. This can be triggered by changing the permission on one of the files to be deleted, for example.

Reply:
What if you create a filerepo/remove folder with mychannel.remove in it, then shut everything down, and then change the permission on filerepo/remove, and then restart? I think this flow down his error path, no?
In a separate test case? and cleanup afterwards?

Reply:
🧹  The missing piece was including execute permissions on the parent directory